storage/bootstrap/result/result_index.go

```diff
@@ -78,6 +79,37 @@ func (r IndexResults) AddResults(other IndexResults) {
 	}
 }
 
+// GetOrAddSegment get or create a new mutable segment.
+func (r IndexResults) GetOrAddSegment(
+	t time.Time,
+	idxopts namespace.IndexOptions,
+	opts Options,
+) (segment.MutableSegment, error) {
+	blockStart := t.Truncate(idxopts.BlockSize())
```

Comment:
maybe add a quick note about why this truncation is required and the `% ==0` guarantee the code relies upon.

Reply:
Sure thing.